src/main/com/aca/homework/week11/map/performance/HighPerformanceMapValueCountCalculator.java

```diff
@@ -0,0 +1,29 @@
+package com.aca.homework.week11.map.performance;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class HighPerformanceMapValueCountCalculator implements MapValueCountCalculator {
+
+    private Map<String, Integer> strings = new HashMap<>();
+
+
+    @Override
+    public StringOccurrence populateAndCalculate(int numberOfRandomStrings) {
+        Randomizer randomizer = new Randomizer();
+        StringOccurrence stringOccurrence = new StringOccurrence();
+
+        for (int i = 0; i < numberOfRandomStrings; i++) {
+            String stringKey = randomizer.get8SizedString();
+            if(strings.containsKey(stringKey)) {
+                int keyCount = strings.get(stringKey) + 1;
+                strings.put(stringKey, keyCount);
+                stringOccurrence.checkMaxMinOccurrence(stringKey, keyCount);
```

Comment:
you are checking this method numberOfRandomStrings times. it is not efficient. instead, please change at the end

Reply:
I need to check each stringKey